src/main/java/uk/gov/hmcts/reform/blobrouter/tasks/processors/BlobProcessor.java

```diff
@@ -60,10 +60,17 @@ public BlobProcessor(
     public void process(BlobClient blobClient) {
         String blobName = blobClient.getBlobName();
         var containerName = blobClient.getContainerName();
+        Instant blobCreationDate = blobClient.getProperties().getLastModified().toInstant();
 
         logger.info("Processing {} from {} container", blobName, containerName);
 
-        envelopeService.saveEvent(containerName, blobName, Event.FILE_PROCESSING_STARTED);
+        UUID id =
+            envelopeService
+                .createNewEnvelope(
```

Comment:
Should we have an audit event when creating a new envelope?

Reply:
Also, this should probably moved after the lease is acquired.

Reply:
`. createNewEnvelope()` service method creates an event